### ISSUE TITLE
Avoid overhead in buffer creation

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2722]
Commit 584369910caf636532b98102ddf51051d2e28241 adds significant overhead in buffer creation that is even visually perceptible (at least on my machine).

Before this commit, `(nyxt:start :failsafe t :urls '("https://github.com/atlas-engineer/nyxt"))` would immediately load the passed URL without flashing into the default startup page.

After this commit, starting Nyxt as above flashes to the default startup page.

My position is that this commit should be reverted, but I offload the final decision to @aartaka and @Ambrevar.

This _could_ explain the impression I've raised in #2720.

I'd also recommend opening a PR when pushing such delicate changes.

**Comment by aartaka** [comment]
Oh, right, `resolve-symbol` in a tight loop. That could explain everything... 

Will revert!

**Comment by aartaka** [comment]
@aadcg, should be fixed in 4e78f4e0384abca1139f5da66cbbe73763e700da!

**Comment by aadcg** [comment]
@aartaka did you try to start Nyxt or load the `nyxt` system after your changes?

```log
; file: /home/aadcg/common-lisp/nyxt/source/mode.lisp
; in: DEFUN RESOLVE-USER-SYMBOL
;     (NSYMBOLS:RESOLVE-SYMBOL NYXT::DESIGNATOR TYPE NYXT::PACKAGES NYXT::ERROR-P)
; 
; caught WARNING:
;   The function SYM:RESOLVE-SYMBOL called with four arguments, but wants at most three.
; 
; caught WARNING:
;   The function SYM:RESOLVE-SYMBOL called with four arguments, but wants at most three.
; 
; caught WARNING:
;   The function SYM:RESOLVE-SYMBOL called with four arguments, but wants at most three.
```

**Comment by aartaka** [comment]
Right, `nsymbols` update with four arguments that I'm using is not yet there...

Fixed in 82c9adaad32c3788ed680b13b97d2e2b25fb26d0!

**Comment by aadcg** [comment]
Commit 82c9adaad32c3788ed680b13b97d2e2b25fb26d0 has a typo on line 232 of the `mode.lisp` file.

**Comment by aartaka** [comment]
So awkward, sorry (ﾉ´ｰ`)ﾉ 

Fixed in 2ae29c20614e887aeff909e5fba06c069c612f60!

**Comment by aadcg** [comment]
All good. Thanks @aartaka.
